src/pyhf/pdf.py

```diff
@@ -211,18 +212,21 @@ def expected_data(self, pars):
             return auxdata[0]
         return auxdata
 
+    def _dataprojection(self, data):
+        tensorlib, _ = get_backend()
+        cut = tensorlib.shape(data)[0] - len(self.config.auxdata)
```

Comment:
is this meant to only return the `auxdata`? I guess you might want to... add a comment on what this function does and test it separately.

Reply:
yes `_dataprojection` is meant to just extract the chunk of data relevant for this subpdf. in #553 this will be replaced with `TensorViewer.split`